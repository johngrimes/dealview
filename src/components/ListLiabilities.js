import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import _ from 'lodash'
import { Link } from 'react-router-dom'

import Breadcrumbs from './Breadcrumbs.js'
import LiabilityActions from '../actions/liabilities.js'

export class ListLiabilities extends React.Component {
  static propTypes = {
    liabilities: PropTypes.shape({
      status: PropTypes.oneOf([ 'uninitialised', 'loading', 'loaded', 'error' ]),
      objects: PropTypes.objectOf(PropTypes.object),
    }),
  }

  constructor(props) {
    super(props)
    if (this.props.liabilities.status === 'uninitialised') {
      this.props.dispatch(LiabilityActions.loadLiabilities())
    }
  }

  breadcrumbs() {
    return [
      { display: 'Portfolio', path: '/portfolio' },
      { display: 'Liabilities', path: '/portfolio/liabilities' },
    ]
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.liabilities.status === 'uninitialised') {
      this.props.dispatch(LiabilityActions.loadLiabilities())
    }
  }

  render() {
    const liabilities = _.map(this.props.liabilities.objects, (v, k) =>
      <li key={k} className='liability'>
        <Link
          className='liability-name'
          to={`/portfolio/liabilities/loans/${v.id}`}
        >
          {v.name}
        </Link>
      </li>
    )
    return (
      <div className='list-liabilities'>
        <Breadcrumbs breadcrumbs={this.breadcrumbs()} />
        <p>
          <Link to='/portfolio/liabilities/new'>New Liability</Link>
        </p>
        <ol className='liabilities'>
          {liabilities}
        </ol>
      </div>
    )
  }
}

const mapStateToProps = state => ({ liabilities: state.liabilities })

export default connect(mapStateToProps)(ListLiabilities)
