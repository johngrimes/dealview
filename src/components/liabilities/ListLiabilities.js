// @flow

import React from 'react'
import type { Children } from 'react'

const ListLiabilities = (props: { children: Children }) => {
  const { children } = props

  return (
    <div className='list-liabilities'>
      <h1>List Liabilities</h1>
      {children}
    </div>
  )
}

export default ListLiabilities
