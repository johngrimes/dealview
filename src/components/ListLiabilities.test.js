import React from 'react'
import { shallow } from 'enzyme'

import { ListLiabilities } from './ListLiabilities.js'
import {
  validLiabilityWithId1,
  validLiabilityWithId2,
} from '../data/fixtures/liability.js'

describe('ListLiabilities', () => {
  it('should render some Liabilities', () => {
    const props = {
      liabilities: {
        status: 'loaded',
        objects: {
          [validLiabilityWithId1.id]: validLiabilityWithId1,
          [validLiabilityWithId2.id]: validLiabilityWithId2,
        },
      },
    }
    const wrapper = shallow(<ListLiabilities {...props} />)
    expect(wrapper.find('.liability').length).toBe(
      Object.keys(props.liabilities).length
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should load Liabilities if uninitialised', () => {
    const dispatch = jest.fn()
    const liabilityActions = require('../actions/liabilities.js').default
    liabilityActions.loadLiabilities = jest.fn()
    const props = {
      liabilities: {
        status: 'uninitialised',
        objects: {},
      },
      dispatch,
    }
    shallow(<ListLiabilities {...props} />)
    expect(dispatch).toHaveBeenCalled()
    expect(liabilityActions.loadLiabilities).toHaveBeenCalled()
  })
})
