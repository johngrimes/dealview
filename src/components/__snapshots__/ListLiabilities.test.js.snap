// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListLiabilities should render some Liabilities 1`] = `
<div
  className="list-liabilities"
>
  <Breadcrumbs
    breadcrumbs={
      Array [
        Object {
          "display": "Portfolio",
          "path": "/portfolio",
        },
        Object {
          "display": "Liabilities",
          "path": "/portfolio/liabilities",
        },
      ]
    }
  />
  <p>
    <Link
      replace={false}
      to="/portfolio/liabilities/new"
    >
      New Liability
    </Link>
  </p>
  <ol
    className="liabilities"
  >
    <li
      className="liability"
    >
      <Link
        className="liability-name"
        replace={false}
        to="/portfolio/liabilities/loans/345"
      >
        5SS Reno Loan
      </Link>
    </li>
    <li
      className="liability"
    >
      <Link
        className="liability-name"
        replace={false}
        to="/portfolio/liabilities/loans/832"
      >
        Briggs Primary
      </Link>
    </li>
  </ol>
</div>
`;
